modules/vsphere-problem-detector-config-checks.adoc

```diff
@@ -0,0 +1,120 @@
+Reconfigure vSphere with a shorter name for the default datastore.
+|icon:check[]
+|icon:check[]
+
+|CheckFolderPermissions
+|Verifies that {product-title} has permission to list volumes in the default datastore.
+This permission is required to create volumes.
+The detector verifies permission by listing the `/` and `/kubevols` directories.
+The root directory must exist.
+It is acceptable if the `/kubevols` directory does not exist when the check runs.
+The directory is created when the datastore is used with dynamic provisioning.
+If this check fails, review the required permissions for the FIXME account at LINK.
+|icon:check[]
+|icon:check[]
+
+|CheckPVs
+|Verifies that the number of characters in the fully qualified path to each persistent volume is less than 255 characters.
+If this check fails, rename the VDMK file to use a shorter name.
+|icon:check[]
+|icon:check[]
+
+|CheckStorageClasses
+a|Verifies the following:
+
+* The names in all storage classes and persistent volumes must be less than 255
+  characters. See FIXME.
+
+* If a storage class uses a storage policy, the storage class must use one policy
+  only and that policy must be defined.
+
+|icon:check[]
+|icon:check[]
+
+|CheckTaskPermissions
+|Verifies that {product-title} is configured with a vCenter account that has privilege to list recent tasks and datastores.
+|icon:check[]
+|icon:check[]
+
+// vsphere_vcenter_info
+|ClusterInfo
+|Collects the cluster version and UUID from vSphere vCenter.
+|icon:check[]
+|icon:check[]
+|===
+
+.Node configuration checks
+[options="header",cols="20,50,^15,^15"]
+|===
+|Name
+|Description
+|Periodic
+|Installation
+
+//CollectNodeESXiVersion - vsphere_esxi_version_total
+//CollectNodeHWVersion - vsphere_node_hw_version_total
+
+|CheckNodeDiskUUID
+|Verifies that all the vSphere virtual machines are configured with `disk.enableUUID=TRUE`.
+FIXME: MOVE THIS. If this check fails, see the link:https://access.redhat.com/solutions/4606201[4606201] support article.
+|icon:check[]
+|icon:times[]
+
+|CheckNodeProviderID
+|Verifies that all nodes are configured with the `ProviderID` from vSphere vCenter.
+This check fails when `get nodes -o json \| jq '.items[]\|[.metadata.name, .spec.providerID]` does not include a value for `.spec.providerID`.
+|icon:check[]
+|icon:times[]
+|===
+
+
+[id="storage-class-configuration-check_{context}"]
+== About the storage class configuration check
+
+The names for persistent volumes that use vSphere storage are related to the datastore name and cluster ID.
+
+When a persistent volume is created, the operating system uses the `systemd` process FIXME: TO DO WHAT, PLEASE?
+The `systemd` process has a 255 character limit for the length of the fully qualified path to the VDMK file that is used for the persistent volume.
+
+The fully qualified path is based on the naming conventions for `systemd` and vSphere.
+The naming conventions use the following pattern:
+
+----
```

Comment:
Should there be a language specification above this source block? I.e.:

`[source,terminal]`